### PR TITLE
feat: Django 4.2 support

### PR DESCRIPTION
**Pull request by fsbraun**
## Description

This PR updates djangocms-moderation to support Django 4.2 and 5.0. It is based on the ideas and work of #256 by @Aiky30 .

* If fixes tests so they can run with django CMS 4.0 and django CMS 4.1
   * CMS 4.0: Tests run with djangocms-version-locking
   * CMS 4.1: Tests run with `DJANGOCMS_VERSIONING_LOCK_VERSIONS` setting set to `True`.

* Since aldryn-forms does not support Django 4.2, the optional `djangocms-moderation.contrib.moderation_forms` app has been discontinued.

* Currently, tests against django CMS 4.0 fail (against django CMS 4.1 everything is fine). 
   @joshyu: This is not an effect of this PR. Tests failed before already. Do you know why? I would love to get them in line.

## Related resources

* #257 
* https://github.com/django-cms/djangocms-versioning/issues/391

## Checklist

* [x] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -22,7 +22,18 @@
...
     classifiers=[
+        'Development Status :: 5 - Production/Stable',
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
+        "Programming Language :: Python :: 3.12",
         "Framework :: Django",
+        "Framework :: Django :: 3.2",
+        "Framework :: Django :: 4.2",
+        "Framework :: Django :: 5.0",
```
Comment (Aiky30) [inline setup.py:33]: Is this tested with 5.0?

File: tests/requirements/requirements_base.txt [hunk 2]
```diff
@@ -9,13 +8,8 @@ django-app-helper
 factory-boy
 flake8
 isort
+ruff
```
Comment (Aiky30) [inline tests/requirements/requirements_base.txt:11]: Should probably keep tooling changes as a separate PR, it's in now so no point wasting time reversing it. I would expect a pyproject.toml file replacing the setup.py file with other config changes for ruff to work correctly.

File: tests/test_monkeypatch.py [hunk 3]
```diff
@@ -76,19 +82,28 @@ def test_get_archive_link(self, _mock):
...
+            version.check_archive.as_bool(self.mock_request.user)
+            archive_link = ""
         # We test that moderation check is called when getting an edit link
         self.assertEqual(1, _mock.call_count)
-        # Edit link is inactive as `is_obj_review_locked` is True
-        self.assertIn("inactive", archive_link)
-        self.assertNotIn(archive_url, archive_link)
+        if versioning_version < "2":
+            # Edit link is inactive as `mock_is_obj_review_locked` is True
+            self.assertIn("inactive", archive_link)
```
Comment (Aiky30) [inline tests/test_monkeypatch.py:97]: Looks to be a regression of functionality in djangocms-versioning 2 where the edit link is disabled vs what appears to have been removed.

**Review by Aiky30 — APPROVED** [review]
Changes look OK here, although the changes on djangocms-versioning regarding the button looks to be regression that may need to be fixed.

File: setup.py (hunk 1 above)
Comment (fsbraun) [reply setup.py:33]: Yes, it is. :-)

**Comment by joshyu** [comment]
@fsbraun , I create a PR https://github.com/django-cms/djangocms-moderation/pull/263 
it fixes all check errors and failed tests.

**Comment by joshyu** [comment]
@fsbraun , about the failed pre-commit-config issue, It is related to the version of isort and other pre-commit plugins.

Freeman and I ever met this errors while fixing another repo.

refer to this: 
https://stackoverflow.com/questions/75269700/pre-commit-fails-to-install-isort-5-11-4-with-error-runtimeerror-the-poetry-co

**Comment by joshyu** [comment]
@fsbraun , 

about the failed tests, it is because that `djangocms-version-locking` addon create a monkeypatch to override `djangocms-moderation.models.ModerationCollection._add_nested_children` method, it add a logic to check whether this version is locked. If locked, current children plugins will not be added to the current collection.

I didn't find when versionlock field in version instance is set, so I hard-code the version check and give the different return value.

**Comment by fsbraun** [comment]
@joshyu Thanks for the explanation. To me, this is very helpful: So that actually is a desired behavior. I also understand https://github.com/FidelityInternational/djangocms-version-locking/issues/80 now (an issue @marksweb raised a while ago)... Let me look into this, maybe we can identical results for CMS 4.0 and CMS 4.1 😃

**Comment by joshyu** [comment]
@fsbraun perhaps need your help to fix the left pre-commit error about 0001_squashed_0017_auto_20220831_0727.py file

**Comment by fsbraun** [comment]
@joshyu Pre-commit now passes. I'd like to look into the version locking effect on the collections with djangocms-versioning 2. Probably need today. Sufficient for you?

**Comment by fsbraun** [comment]
@joshyu I've updated the tests to test the same outcomes for djangocms-versioning 1 and 2. For most tests that means, removing the version lock before the test. I've also added a test that ensures that locked versions are not added to collections. If you're ok, I can merge.

**Comment by joshyu** [comment]
@fsbraun ,
LGTM. Thank you for giving such a perfect solution, this is really better than my update on `test_view.py`.
please help to merge it.

Thanks.
